api/v1beta1/gcpmachine_types.go

```diff
@@ -66,6 +66,41 @@ const (
 	IPForwardingDisabled IPForwarding = "Disabled"
 )
 
+// AcceleratorType is a type to use to define which accelerator type will be used.
+type AcceleratorType string
+
+const (
+	// All supported GPU accelerator types.
+	NVIDIA_K80_GPUS AcceleratorType = "nvidia-tesla-k80"
+	NVIDIA_P100_GPUS AcceleratorType = "nvidia-tesla-p100"
+	NVIDIA_V100_GPUS AcceleratorType = "nvidia-tesla-v100"
+	NVIDIA_A100_GPUS AcceleratorType = "nvidia-tesla-a100"
+	NVIDIA_P4_GPUS AcceleratorType = "nvidia-tesla-p4"
+	NVIDIA_T4_GPUS AcceleratorType = "nvidia-tesla-t4"
+)
+
+// AcceleratorConfig is the GPU accelerator configuration for the GCP machine.
+type AcceleratorConfig struct {
+	// Type is the type of the GPU accelerator to be used for the GCP machine.
+	// +kubebuilder:required
+	// +kubebuilder:validation:Enum
+	Type AcceleratorType `json:"type"`
+
+	// Count is the number of accelerators to be used for the GCP machine.
+	// +kubebuilder:default:=1
+	// +kubebuilder:validation:Minimum=1
+	Count int `json:"acceleratorCount,omitempty"`
```

Comment:
https://pkg.go.dev/google.golang.org/api@v0.86.0/compute/v1#AcceleratorConfig

here the count is int64 should we make it int > int64?

Reply:
Otherwise, I guess we might have to convert everything back again while passing these values while reconciliation.

Reply:
https://pkg.go.dev/google.golang.org/api@v0.86.0/compute/v1#Scheduling.AutomaticRestart

The AutomaticRestarts accepts a `*bool` should we also change this?

Reply:
```suggestion
	Count int `json:"count,omitempty"`
```

Reply:
`int` and `bool` is ok we just have to convert while passing
we should focusing on the actual logic 😃